dynawo/sources/ModelicaCompiler/DYNCompileModelicaModel.cpp

```diff
@@ -450,16 +450,17 @@ compileLib(const string& modelName, const string& compilationDir) {
 }
 
 bool verifySharedObject(const string& library) {
-  // dlopen include <dlfcn.h>: to see if a shared object file
-  const char* filename = library.c_str();
-  void *handle;
-  handle = dlopen(filename, RTLD_NOW);
-  if (!handle) {
-    fprintf(stderr, "%s\n", dlerror());
-    printf(" DYNCompileModelicaModel: could not open .so by dlopen.");
+  try {
+    boost::dll::shared_library lib(library);
+    static_cast<void>(lib);
+    // we don't use the lib as we check that the library is loadable, which is done in
+    // constructor
+  }
+  catch(const std::exception& e) {
+    std::cerr << e.what() << '\n';
```

Comment:
std::endl inseatd of '\n'

Reply:
artefact from previous code. will be modified